apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -177,12 +166,14 @@ export class CredentialsService implements CredentialVerifier {
     vp: VerifiablePresentationDto,
     options: VerifyOptionsDto
   ): Promise<VerificationResultDto> {
-    const verifyOptions: ISpruceVerifyOptions = options;
-
-    return didKitExecutor<VerificationResultDto>(
-      () => verifyPresentation(JSON.stringify(vp), JSON.stringify(verifyOptions)),
-      'verifyPresentation'
+    const w3cVerifyPresentationOptions: W3cVerifyPresentationOptions = {
+      presentation: JsonTransformer.fromJSON(vp, W3cJsonLdVerifiablePresentation),
+      challenge: options.challenge
```

Comment:
I think it would be good to have the options.challenge be an override but typically one can just refer to the challenge in the proof.
```suggestion
      challenge: options.challenge ?? vp.proof.challenge 
```
Note that this required changing the `VerifiablePresentationDto` `proof` property type:
```
proof: Record<string, unknown> & {
    challenge?: string;
  };
```